src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -6,49 +6,55 @@ namespace ray {
 
 Status CoreWorkerDirectTaskSubmitter::SubmitTask(TaskSpecification task_spec) {
   resolver_.ResolveDependencies(task_spec, [this, task_spec]() {
-    // TODO(ekl) should have a queue per distinct resource type required
     absl::MutexLock lock(&mu_);
-    queued_tasks_.push_back(task_spec);
-    RequestNewWorkerIfNeeded(task_spec);
+    const TaskQueueKey queue_key(task_spec.GetSchedulingClass(),
+                                 task_spec.GetDependencies());
```

Comment:
Does this handle the case correctly where the dependency resolution inlines args by mutating the task spec?